CMakeLists.txt

```diff
@@ -340,49 +337,9 @@ set(scylla_sources
     transport/event_notifier.cc
     transport/messages/result_message.cc
     transport/server.cc
-    types.cc
```

Comment:
This is not some random utility, this is the implementation of the CQL type system. This should be moved to `types/`.

Reply:
i will defer the change to a follow-up commit. we do need to fix it.

Reply:
see #12926